model.go

```diff
@@ -0,0 +1,89 @@
+package kallax
+
+import (
+	"database/sql"
+	"database/sql/driver"
+
+	uuid "github.com/satori/go.uuid"
+)
+
+type ID uuid.UUID
+
+func (id *ID) Scan(src interface{}) error {
+	return (*uuid.UUID)(id).Scan(src)
+}
+
+func (id ID) Value() (driver.Value, error) {
+	return uuid.UUID(id).Value()
+}
+
+var _ sql.Scanner = (*ID)(nil)
+var _ driver.Valuer = (*ID)(nil)
+
+func NewID() ID {
+	return ID(uuid.NewV4())
+}
+
+func (id ID) IsEmpty() bool {
+	return uuid.UUID(id) == uuid.Nil
+}
+
+type Model struct {
+	ID        ID
+	persisted bool
+}
+
+func (m *Model) GetID() ID {
+	return m.ID
+}
+
+func (m *Model) SetID(id ID) {
+	m.ID = id
+}
+
+func (m *Model) IsPersisted() bool {
+	return m.persisted
+}
+
+func (m *Model) setPersisted(isPersisted bool) {
+	m.persisted = isPersisted
```

Comment:
I would say that something that is persisted cannot be set as not persisted, so maybe just `setPersisted` without params.